### PR TITLE
refactor actors cache to it for memory+redis one

### PR DESCRIPTION
**Pull request by emmanuelm41**
:link: [zboto Link](https://app.clickup.com/t/8693dkamz)

### CODE REVIEW

File: actors/cache/impl/zcache.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+func (m *ZCache) storeRobustShort(robust string, short string) {
+	if robust == "" || short == "" {
+		// zap.S().Warn("[ActorsCache] - Trying to store empty robust or short address")
+		return
+	}
+
+	ctx := context.Background()
+	_ = m.robustShortMap.Set(ctx, robust, short, -1)
```
Comment (lucaslopezf) [inline actors/cache/impl/zcache.go:184]: It's unclear what -1 signifies. We could use a named constant to make this more readable (and in the other parts that -1 appears)

File: actors/cache/impl/zcache.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+func (m *ZCache) BackFill() error {
+	// Nothing to do
+	return nil
+}
+
+func (m *ZCache) GetActorCode(address address.Address, key filTypes.TipSetKey) (string, error) {
+	shortAddress, err := m.GetShortAddress(address)
+	if err != nil {
+		m.logger.Sugar().Infof("[ActorsCache] - Error getting short address: %s\n", err.Error())
```
Comment (lucaslopezf) [inline actors/cache/impl/zcache.go:109]: The level of log is Info but the message is Error, is it ok?

File: actors/cache/impl/zcache.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+		return cid.Undef.String(), common.ErrKeyNotFound
```
Comment (lucaslopezf) [inline actors/cache/impl/zcache.go:110]: Adding context about which key was not found could be helpful for debugging

**Review by lucaslopezf — COMMENTED** [review]
Just a few comments

File: actors/cache/impl/zcache.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+	if shortAdd == "" {
+		return "", common.ErrEmptyValue
+	}
+
+	return shortAdd, nil
+}
+
+func (m *ZCache) storeRobustShort(robust string, short string) {
+	if robust == "" || short == "" {
+		// zap.S().Warn("[ActorsCache] - Trying to store empty robust or short address")
```
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go:179]: I am not sure. It was there before this commit haha

File: actors/cache/impl/zcache.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+		return
+	}
+
+	ctx := context.Background()
+	_ = m.robustShortMap.Set(ctx, robust, short, -1)
+}
+
+func (m *ZCache) storeShortRobust(short string, robust string) {
+	if robust == "" || short == "" {
+		// zap.S().Warn("[ActorsCache] - Trying to store empty robust or short address")
```
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go:189]: I am not sure. It was there before this commit haha

File: actors/cache/impl/zcache.go (hunk 2 above)
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go:109]: Again, it was there before this PR ahaha

File: actors/cache/impl/zcache.go (hunk 3 above)
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go:110]: I like it!

File: actors/cache/impl/zcache.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go]: Done!

File: actors/cache/impl/zcache.go [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+	var err error
+	m.logger = logger2.GetSafeLogger(logger)
+
+	// If no config was provided, the combined cache is configured as
+	// remote best effort, as the remote cache will fail. However, the cache will
+	// work anyway
+	cacheConfig := source.Config.Cache
+	if cacheConfig == nil {
+		m.cacheType = "in-memory"
+		if m.shortCidMap, err = zcache.NewLocalCache(&zcache.LocalConfig{Prefix: Short2CidMapPrefix}); err != nil {
```
[the remark this answers is not in the document]
Comment (emmanuelm41) [reply actors/cache/impl/zcache.go]: Done!

**Pull request merged**